### PR TITLE
docs(bitbucket-pipelines): authenticated registries

### PR DESCRIPTION
**Pull request by setchy**
## Changes

Add documentation for
- authenticated registry support
- migrating from unauthenticated to authenticated registry formats

## Context

Publishing examples from these two sources
- https://github.com/renovatebot/renovate/issues/13555#issuecomment-1479419939
- https://github.com/renovatebot/renovate/discussions/21094#discussioncomment-5399664

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
we should extend manager support instead

**Comment by setchy** [comment]
> we should extend manager support instead 

Agree, and there is an open issue to track that.  

But until then, why not improve the docs? Can easily remove the regexs manager examples if and when the bitbucket-pipelines manager parses the yml, which would only improve authenticated image handling, I cannot see it supporting moving from unauthenticated to authenticated image structures

**Comment by setchy** [comment]
> we should extend manager support instead 

Agree, and there is an open issue to track that.  

But until then, why not improve the docs? Can easily remove the regexs manager examples if and when the bitbucket-pipelines manager parses the yml, which would only improve authenticated image handling, I cannot see it supporting moving from unauthenticated to authenticated image structures

**Comment by setchy** [comment]
> we should extend manager support instead

https://github.com/renovatebot/renovate/pull/21107 - updated bitbucket-pipelines manager extract logic. 

_Note: I have not attempted to parse the yaml as suggested by @viceice [here](https://github.com/renovatebot/renovate/issues/13555#issuecomment-1012905810) - so ordering is important (but also incredibly common)._

**Comment by setchy** [comment]
Moving to draft until feedback re: https://github.com/renovatebot/renovate/pull/21107

File: lib/modules/manager/bitbucket-pipelines/readme.md [hunk 1]
```diff
@@ -1,3 +1,50 @@
 Extracts dependencies from Bitbucket Pipelines config files.
 
 If you need to change the versioning format, read the [versioning](https://docs.renovatebot.com/modules/versioning/) documentation to learn more.
+
+### Using private build images
+
+Bitbucket Pipelines supports using an authenticated Docker registry, including self-hosted private registries, Docker Hub, Amazon ECR and Google GCR.
+
+More details can be found on the Atlassian [Bitbucket Pipeline documentation](https://support.atlassian.com/bitbucket-cloud/docs/use-docker-images-as-build-environments) site
```
Comment (HonkingGoose) [inline lib/modules/manager/bitbucket-pipelines/readme.md:9]:
```suggestion
More details can be found on the Atlassian [Bitbucket Pipeline documentation](https://support.atlassian.com/bitbucket-cloud/docs/use-docker-images-as-build-environments) site.
```

File: lib/modules/manager/bitbucket-pipelines/readme.md [hunk 2]
```diff
@@ -1,3 +1,50 @@
...
+
+The following `regexManager` configuration can be used for authenticated Docker Hub or private registries. Changes will be required for Amazon ECR or Google GCR use cases.
```
Comment (HonkingGoose) [inline lib/modules/manager/bitbucket-pipelines/readme.md:11]:
```suggestion
The following `regexManager` configuration can be used for authenticated Docker Hub or private registries.
You'll need to change this example config if you're using Amazon ECR or Google GCR.
```

Also: do our readers know _what_ changes they'll need to make if they're using Amazon ECR or Google GCR? I don't see this explained in the text, but maybe we can assume our readers are competent with ECR/GCR already?

File: lib/modules/manager/bitbucket-pipelines/readme.md [hunk 3]
```diff
@@ -1,3 +1,50 @@
...
+    }
+  ]
+}
+```
+
+### Replacing unauthenticated with authenticated registries
+
+With the [regex manager](https://docs.renovatebot.com/modules/manager/regex/) you can configure Renovate to assist with migrating your Bitbucket Pipelines from unauthenticated registries to authenticated registries.
+
+An example configuration is shown below
```
Comment (HonkingGoose) [inline lib/modules/manager/bitbucket-pipelines/readme.md:32]:
```suggestion
For example:
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/modules/manager/bitbucket-pipelines/readme.md (hunk 1 above)
Comment (setchy) [reply lib/modules/manager/bitbucket-pipelines/readme.md:9]: Thanks @HonkingGoose 

I'll be updating this PR one @viceice can land https://github.com/renovatebot/renovate/pull/21102

**Pull request closed without merging**